### ISSUE TITLE
Vibration not work on my device Samsung Galaxy A6 (Android version 9 / Pie)

### ISSUE THREAD

**Issue by rrifafauzikomara** [flutter/flutter#56534]
### Steps to Reproduce

1. Add `package:flutter/services.dart` and `import 'dart:io` to the code.

2. Add an example to try `vibrartion`
```
import 'package:flutter/material.dart';
import 'package:flutter/services.dart'; // we need this for the vibrations
import 'dart:io'; // we need this for the sleep method

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Vibrate Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: VibrateHomepage(),
    );
  }
}

class VibrateHomepage extends StatefulWidget {
  VibrateHomepage({Key key}) : super(key: key);

  @override
  _VibrateHomepageState createState() => _VibrateHomepageState();
}

class _VibrateHomepageState extends State<VibrateHomepage> {
  _PatternVibrate() {
    HapticFeedback.mediumImpact();

    sleep(
      const Duration(milliseconds: 200),
    );

    HapticFeedback.mediumImpact();

    sleep(
      const Duration(milliseconds: 500),
    );

    HapticFeedback.mediumImpact();

    sleep(
      const Duration(milliseconds: 200),
    );
    HapticFeedback.mediumImpact();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.stretch,
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Expanded(
              child: FlatButton(
                child: Text('HapticFeedback.vibrate()'),
                onPressed: () {
                  HapticFeedback.vibrate();
                },
                color: Colors.deepPurple[100],
              ),
            ),
            Expanded(
              child: FlatButton(
                child: Text('HapticFeedback.selectionClick()'),
                onPressed: () {
                  HapticFeedback.selectionClick();
                },
                color: Colors.deepPurple[200],
              ),
            ),
            Expanded(
              child: FlatButton(
                child: Text('HapticFeedback.lightImpact()'),
                onPressed: () {
                  HapticFeedback.lightImpact();
                },
                color: Colors.deepPurple[300],
              ),
            ),
            Expanded(
              child: FlatButton(
                child: Text('HapticFeedback.mediumImpact()'),
                onPressed: () {
                  HapticFeedback.mediumImpact();
                },
                color: Colors.deepPurple[400],
              ),
            ),
            Expanded(
              child: FlatButton(
                child: Text('HapticFeedback.heavyImpact()'),
                onPressed: () {
                  HapticFeedback.heavyImpact();
                },
                color: Colors.deepPurple[500],
              ),
            ),
            Expanded(
              child: FlatButton(
                child: Text('Pattern - heart beat'),
                onPressed: () {
                  _PatternVibrate();
                },
                color: Colors.deepPurple[600],
              ),
            )
          ],
        ),
      ),
    );
  }
}
```
3. Add permission on `AndroidManifest.xml` (main)
```
<uses-permission android:name="android.permission.VIBRATE" />
```
4. Run the Apps (But `vibration` not working on my `device`)
5. Here my flutter doctor result
```
[✓] Flutter (Channel stable, v1.12.13+hotfix.8, on Mac OS X 10.15.2 19C57,
    locale en-US)
    • Flutter version 1.12.13+hotfix.8 at
      /Users/rrifafauzikomara/Documents/flutter
    • Framework revision 0b8abb4724 (3 months ago), 2020-02-11 11:44:36 -0800
    • Engine revision e1e6ced81d
    • Dart version 2.7.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/rrifafauzikomara/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling
      support)
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4.1, Build version 11E503a
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.39.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.5.1

[✓] Connected device (1 available)
    • SM A600G • 5200e3e7f01e45f3 • android-arm • Android 9 (API 28)

• No issues found!
```

**Comment by TahaTesser** [comment]
Hi @rrifafauzikomara 
Does it work on work on other device? 
can you please provide your `flutter run --verbose` while running the code and tapping? 
Thank you

**Comment by rrifafauzikomara** [comment]
> Hi @rrifafauzikomara
> Does it work on work on other device? and make sure you've vibration enabled
> can you please provide your `flutter run --verbose` while running the code and tapping?
> Thank you

Hi @TahaTesser 
I don't know because I only have one device (Samsung A6 with Android version 9 / Pie) and what the mean of `and make sure you've vibration enabled`?
So I just run the apps with run `flutter run --verbose` and paste the log here?

**Comment by rrifafauzikomara** [comment]
@TahaTesser 

This is the result of `flutter run --verbose` while running the code and tapping

```
[   +2 ms] 🔥  To hot reload changes while running, press "r". To hot restart (and rebuild state), press "R".
[   +1 ms] An Observatory debugger and profiler on SM A600G is available at: http://127.0.0.1:61583/qkq2UmsAQBc=/
[        ] For a more detailed help message, press "h". To detach, press "d"; to quit, press "q".
[+3369 ms] D/ViewRootImpl@fcdff38[MainActivity](14140): ViewPostIme pointer 0
[  +98 ms] D/ViewRootImpl@fcdff38[MainActivity](14140): ViewPostIme pointer 1
[+4600 ms] D/ViewRootImpl@fcdff38[MainActivity](14140): ViewPostIme pointer 0
[  +92 ms] D/ViewRootImpl@fcdff38[MainActivity](14140): ViewPostIme pointer 1
[ +752 ms] D/ViewRootImpl@fcdff38[MainActivity](14140): ViewPostIme pointer 0
[ +151 ms] D/ViewRootImpl@fcdff38[MainActivity](14140): ViewPostIme pointer 1
[ +967 ms] D/ViewRootImpl@fcdff38[MainActivity](14140): ViewPostIme pointer 0
[  +99 ms] D/ViewRootImpl@fcdff38[MainActivity](14140): ViewPostIme pointer 1
[ +604 ms] D/ViewRootImpl@fcdff38[MainActivity](14140): ViewPostIme pointer 0
[ +117 ms] D/ViewRootImpl@fcdff38[MainActivity](14140): ViewPostIme pointer 1
[ +570 ms] D/ViewRootImpl@fcdff38[MainActivity](14140): ViewPostIme pointer 0
[  +81 ms] D/ViewRootImpl@fcdff38[MainActivity](14140): ViewPostIme pointer 1
[+1207 ms] D/ViewRootImpl@fcdff38[MainActivity](14140): ViewPostIme pointer 0
[  +67 ms] D/ViewRootImpl@fcdff38[MainActivity](14140): ViewPostIme pointer 1
```

**Comment by rrifafauzikomara** [comment]
I found the solution, this is my phone hardware problem.
